src/main/resources/XWiki/SuggestDisplayEditItems.xml

```diff
@@ -0,0 +1,1023 @@
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <name>XWiki.SuggestDisplayEditItems</name>
+    <number>1</number>
+    <className>XWiki.JavaScriptExtension</className>
+    <guid>6afc2baf-1884-49fa-a64f-94053bc12537</guid>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery'],function($){
+$.extend({
+	createUploadIframe: function(id, uri){
+		//create frame
+		var frameId = 'jUploadFrame' + id;
+		var iframeHtml = '&lt;iframe id="' + frameId + '" name="' + frameId + '" style="position:absolute; top:-9999px; left:-9999px"';
+		if(window.ActiveXObject){
+			if(typeof uri== 'boolean'){
+				iframeHtml += ' src="' + 'javascript:false' + '"';
+			}
+			else if(typeof uri== 'string'){
+				iframeHtml += ' src="' + uri + '"';
+			}	
+		}
+		iframeHtml += ' /&gt;';
+		$(iframeHtml).appendTo(document.body);
+		return $('#' + frameId).get(0);			
+    },
+    createUploadForm: function(id, fileElementId, data){
+		//create form	
+		var formId = 'jUploadForm' + id;
+		var fileId = 'jUploadFile' + id;
+		var form = $('&lt;form  action="" method="POST" name="' + formId + '" id="' + formId + '" enctype="multipart/form-data"&gt;&lt;/form&gt;');	
+		if(data){
+			for(var i in data)
+			{
+				$('&lt;input type="hidden" name="' + i + '" value="' + data[i] + '" /&gt;').appendTo(form);
+			}			
+		}		
+		var oldElement = $('#' + fileElementId);
+		var newElement = $(oldElement).clone();
+		$(oldElement).attr('id', fileId);
+		$(oldElement).before(newElement);
+		$(oldElement).appendTo(form);
+		
+		//set attributes
+		$(form).css('position', 'absolute');
+		$(form).css('top', '-1200px');
+		$(form).css('left', '-1200px');
+		$(form).appendTo('body');		
+		return form;
+    },
+	ajaxFileUpload: function(s) {
+        // TODO introduce global settings, allowing the client to modify them for all requests, not only timeout		
+        s = $.extend({}, $.ajaxSettings, s);
+        var id = new Date().getTime()        
+		var form = $.createUploadForm(id, s.fileElementId, (typeof(s.data)=='undefined'?false:s.data));
+		var io = $.createUploadIframe(id, s.secureuri);
+		var frameId = 'jUploadFrame' + id;
+		var formId = 'jUploadForm' + id;		
+        // Watch for a new set of requests
+        if ( s.global &amp;&amp; ! $.active++ ){
```

Comment:
Same thing, this method is from the jQuery plugin
http://www.phpletter.com/Our-Projects/AjaxFileUpload/